core/classes/IssueFileAddedTimelineEvent.class.php

```diff
@@ -0,0 +1,58 @@
+<?php
+# MantisBT - A PHP based bugtracking system
+
+# MantisBT is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# MantisBT is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with MantisBT.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Timeline event class for adding files to issues.
+ * @copyright Copyright 2014 MantisBT Team - mantisbt-dev@lists.sourceforge.net
+ * @link http://www.mantisbt.org
+ * @package MantisBT
+ */
+
+/**
+ * Timeline event class for creation of issues.
+ *
+ * @package MantisBT
+ * @subpackage classes
+ */
+class IssueFileAddedTimelineEvent extends TimelineEvent {
+	private $issue_id;
+
+	/**
+	 * @param integer $p_timestamp Timestamp representing the time the event occurred.
+	 * @param integer $p_user_id   An user identifier.
+	 * @param integer $p_issue_id  A issue identifier.
+	 */
+	public function __construct( $p_timestamp, $p_user_id, $p_issue_id ) {
+		parent::__construct( $p_timestamp, $p_user_id );
+
+		$this->issue_id = $p_issue_id;
+	}
+
+	/**
+	 * Returns html string to display
+	 * @return string
+	 */
+	public function html() {
+		$t_show_summary = config_get( 'timeline_show_issue_summary' );
```

Comment:
This throws `APPLICATION WARNING #100: Configuration option "timeline_show_issue_summary" not found.` , which is a new feature added in #1127 (still under review/discussion); I would suggest to keep the 2 PRs independent from each other.

See also https://github.com/mantisbt/mantisbt/pull/1127#discussion_r130890947